FactoryDesignPattern.java

```diff
@@ -0,0 +1,97 @@
+import java.util.Scanner;
+
+public class FactoryDesignPattern {
+    public static void main(String[] args) {
+        Logistics l= new Logistics();
+        try {
+            l.dealWithLogistic();
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+        }
+
+    }
+}
+
+class Logistics {
+    void dealWithLogistic() throws Exception{
+        Scanner sc= new Scanner(System.in);
```

Comment:
Scanner object needs to be closed after use.

Reply:
Done